### DISCUSSION TITLE
HasManyThrough  with unlimited intermediate models.

### DISCUSSION THREAD

**Discussion by meideros** [MasoniteFramework/orm#493] [Ideas]
I realize that orator orm does not allow us to apply hasmanythrough between more than three models, and since masonite orm is a package that aims to replace orator orm, I suggest that this be possible with masinite orm.

**Comment by josephmancuso** [comment]
Transferring this to the orm repo

**Comment by josephmancuso** [comment]
@meideros how do you expect something like this to look? This would require a huge and confusing chain of primary keys and models.

If these relationships are too distant it might make more sense from a database perspective to denormalize a bit

**Comment by josephmancuso** [comment]
It would have to work like this

https://github.com/staudenmeir/eloquent-has-many-deep

**Comment by jpmateo022** [comment]
> It would have to work like this
> 
> https://github.com/staudenmeir/eloquent-has-many-deep

This is actually a good feature to add sometimes I have this kind of scenario in my app.

**Comment by meideros** [comment]
Exactly it will look like this  
https://github.com/staudenmeir/eloquent-has-many-deep

**Comment by projectmentor** [comment]
Guys I think this is the right place to share another very good breakdown of this requirement. I constantly refer back to this article in  my Laravel development. Hope it helps.
https://p.softonsofa.com/laravel-querying-any-level-far-relations-with-simple-trick/

**Comment by josephmancuso** [comment]
If I have some extra time on my hands I can work on this but it's not something I can get to soon. I am willing to test and review any pull requests that attempt to solve this

**Comment by projectmentor** [comment]
I'm just not that good in Python to be of much help, but I think that the  "trick" in the above example is passing a reference (pointer) back from the lambda (?) function  which helps to avoid a great deal of effort otherwise. hope I'm using the correct python terms

**Comment by josephmancuso** [comment]
@projectmentor yeah that might actually work. Thats a nice little workout. I'd have to try it to see if it works here

**Comment by josephmancuso** [comment]
Moved this to a discussion to more accurately reflect the status of this
